### ISSUE TITLE
Notifications - user friendly features

### ISSUE THREAD

**Issue by mzebrak** [Textualize/textual#3146]
I really love the textual notifications. We already had some basic implementation of this feature before `Textual` included it, so we quickly replaced it with the official feature.

But what I'm actually missing are a few features that will help when the notification stack gets full to the top.

Consider this example - notifications about "errors" are quite important, so we don't want a situation where the user can't see such a message. For this reason, we disable the timeout, we want the user to manually get rid of such notification.

```python
import math

from textual.app import App, ComposeResult
from textual.widgets import Placeholder


class MyApp(App[None]):

    def on_mount(self) -> None:
        for _ in range(20):
            self.notify("Something bad happened!", severity="error", timeout=math.inf)

    def compose(self) -> ComposeResult:
        yield Placeholder("Hello, world!")


if __name__ == "__main__":
    MyApp().run()
```

In my opinion things like these would be very useful and will help with the situation above (in the order they seem most necessary):
1. Make notification stack scrollable
2. Add some button(?) to dismiss all the notifications
3. Add some filtering mechanism so they can be filtered by severity
4. Add some place to see the notification history?

I'm looking forward to your reply - what do you think?

**Comment by davep** [comment]
> Add some button(?) to dismiss all the notifications

Just as a passing comment, and while not keyboard-friendly of course: clicking on a notification with the mouse will dismiss it.

**Comment by mzebrak** [comment]
> > Add some button(?) to dismiss all the notifications
> 
> Just as a passing comment, and while not keyboard-friendly of course: clicking on a notification with the mouse will dismiss it.

Yeah, but in the mentioned example, you have to click 20 times to dismiss all the notifications. I rather mean quick action to dismiss **all** the notifications at once.

**Comment by davep** [comment]
As I said: just a passing comment, not a negation of any particular request or suggestion. :-)
